### DISCUSSION TITLE
Edit Procedure

### DISCUSSION THREAD

**Discussion by Brummi1980** [dbeaver/dbeaver#13660] [Q&A]
hello dear community ...

I have, let's say, a medium problem. We had a programmer create an individual solution for calculating commission. Unfortunately the programmer let us sit and is no longer available. Now we have the problem that we have to change a few values ​​in the calculation.

Here is the corresponding range the values ​​for LTURNOVER have to adjust to higher values:

```
- Calculation of bonus commission
  IF (LTURNOVER> = 5000.0) THEN
    LCOMMISSION_BONUS = 75.0;
  ELSE IF (LTURNOVER> = 4000.0) THEN
    LCOMMISSION_BONUS = 50.0;
  ELSE IF (LTURNOVER> = 3000.0) THEN
    LCOMMISSION_BONUS = 35.0;
  ELSE IF (LTURNOVER> = 2000.0) THEN
    LCOMMISSION_BONUS = 20.0;
  ELSE
    LCOMMISSION_BONUS = 0;
```

![image](https://user-images.githubusercontent.com/89509849/130745133-b35033e6-7d5e-4935-8b19-9501ec7440b1.png)

DBeaver version 3.7.8 is installed on our computer (Win 10 Pro)

How can I change the values ​​here. Is this only possible via the console or can it be done directly with DBeaver.

Before making a change, I would of course like to save the old procedure in case something goes wrong. How can I duplicate / rename the procedure in _old?

I hope you can help me and the information provided is enough to find a solution.

I hope for the help of the community.

**Comment by LonwoLonwo** [comment]
Hello @Brummi1980 

The ability to edit procedure body in FireBird was implemented in DBeaver version 21.0.1
https://github.com/dbeaver/dbeaver/issues/10872

You can test it in our latest version: https://dbeaver.io/
Or choose another version here: https://dbeaver.io/files/

Procedure duplication via UI is not supported. You can copy the procedure body and create duplicate in SQL Editor.

**Reply by Brummi1980** [reply]
Hello  @LonwoLonwo,

many thanks for your response. Can I update DBeaver without losing the connection setting? 

How can I create a duplicate in SQL Editor. It is possible to rename the current procedure in ...._old

**Reply by LonwoLonwo** [reply]
Connection settings information is stored in your workspace, so settings should be preserved.

![2021-08-25 13_31_43-DBeaver 21 1 0 - numbers](https://user-images.githubusercontent.com/45152336/130775155-54f2d48b-33a6-4c50-859f-6c0034dd6c58.png)

> How can I create a duplicate in SQL Editor. It is possible to rename the current procedure in ...._old

Just open the new SQL Editor, paste your procedure body there and add "_old" to the procedure name. 

![2021-08-25 13_34_57-DBeaver 21 1 0 - numbers](https://user-images.githubusercontent.com/45152336/130775653-5fa9f6d7-b696-4db1-b681-6d3a9c1da1af.png)

![2021-08-25 13_35_27-DBeaver 21 1 0 - _none_ Script-37](https://user-images.githubusercontent.com/45152336/130775656-e00b6f61-b639-425c-97f3-f19920e2d7ed.png)

**Reply by Brummi1980** [reply]
Great, it works. Thank you very much.
